File: src/main/java/ca/mcgill/science/tepid/client/ui/full/notification/NotificationTest.java
package ca.mcgill.science.tepid.client.ui.full.notification;

public class NotificationTest {

    public static void main(String[] args) throws InterruptedException {
        String jobName = "blah_blah_blah.docx - Microsoft Word";
        Notification n = new Notification();
        n.setStatus(0x2196F3, "receiving", "Your job is uploading", "Your print job \"" + jobName + "\" is currently being received from the application. ");
        n.setVisible(true);
        Thread.sleep(3000);
        n.setStatus(0x4D983E, "sending", "Sending job to printer", "\"" + jobName + "\" has processed and is being sent to 1B18-Southpole.");
        Thread.sleep(3000);
//        Thread.sleep(2000);
//        n.setQuota(15, 103, "Topping up your quota","Kaching kaching");
//        n.setStatus(0xB71C1C, "color", "Cannot print colour job", "\"blah_blah_blah.docx - Microsoft Word\" will not print because colour printing is not enabled for your account.");
//        Notification n2 = new Notification();
//        n2.setStatus(0x2196F3, "receiving", "Your job is uploading", "Your print job \"blah_blah_blah.docx - Microsoft Word\" is currently being received from the application. ");
//        n2.setVisible(true);
//        Thread.sleep(4000);
//        n.setStatus(0xB71C1C, "noquota", "Insufficient balance", "We cannot print \"blah_blah_blah.docx - Microsoft Word\" (122 pages) because you only have 17 pages left this term.");
//        Thread.sleep(2000);
//        n2.setStatus(0xB71C1C, "fail", "Failed to print", "Your job \"blah_blah_blah.docx - Microsoft Word\" failed during processing, please contact CTF.");
//        Thread.sleep(4000);
//        n.setStatus(0x4D983E, "sending", "Sending job to printer", "\"blah_blah_blah.docx - Microsoft Word\" has processed and is being sent to 1B18-Right. ");
//        n2.setQuota(88, 22, "Losing quota :(","Kaching kaching?");
//        Thread.sleep(4000);
        if (n.isClosed()) {
            n = new Notification();
            n.setVisible(true);
        }
        final int credits = 19, creditsBefore = 71;
        n.setQuota(creditsBefore, credits, "You have " + credits + " pages left", "\"blah_blah_blah.docx - Microsoft Word\" sent to printer 1B18-Right.");
        Thread.sleep(6000);
        n.setStatus(0x2196F3, "receiving", "Your job is uploading", "Your print job isn't actually uploading ;) ");
    }

}

File: src/test/java/ca/mcgill/science/tepid/client/ui/NotificationTest.kt
package ca.mcgill.science.tepid.client.ui

import ca.mcgill.science.tepid.client.ui.notification.*
import org.junit.Test
import java.awt.Color
import kotlin.test.assertEquals

class NotificationTest {

    @Test
    fun test() {
        val jobName = "blah_blah_blah.docx - Microsoft Word"
        var n = NotificationOld()
        n.setStatus(0x2196F3, "receiving", "Your job is uploading", "Your print job \"$jobName\" is currently being received from the application. ")
        n.isVisible = true
        Thread.sleep(3000)
        n.setStatus(0x4D983E, "sending", "Sending job to printer", "\"$jobName\" has processed and is being sent to 1B18-Southpole.")
        Thread.sleep(3000)
        if (n.isClosed) {
            n = NotificationOld()
            n.isVisible = true
        }
        val credits = 19
        val creditsBefore = 71
        n.setQuota(creditsBefore, credits, "You have $credits pages left", "\"blah_blah_blah.docx - Microsoft Word\" sent to printer 1B18-Right.")
        Thread.sleep(6000)
        n.setStatus(0x2196F3, "receiving", "Your job is uploading", "Your print job isn't actually uploading ;) ")
    }

    @Test
    fun test2() {
        val jobName = "blah_blah_blah.docx - Microsoft Word"
        Notifications.notify(StateNotification("Your job is uploading", "Your print job \"$jobName\" is currently being received from the application.", 0x2196F3, NotificationIcon.RECEIVING))
        Thread.sleep(3000)
        Notifications.notify(StateNotification("Sending job to printer", "\"$jobName\" has processed and is being sent to 1B18-Southpole.", 0x4D983E, NotificationIcon.SENDING))
        Thread.sleep(3000)
        val credits = 19
        val creditsBefore = 71
        Notifications.notify(TransitionNotification("You have $credits pages left", "\"blah_blah_blah.docx - Microsoft Word\" sent to printer 1B18-Right.", creditsBefore, credits))
        Thread.sleep(6000)
        Notifications.notify(StateNotification("Your job is uploading", "Your print job isn't actually uploading ;) ", 0x2196F3, NotificationIcon.RECEIVING))
        Notifications.thread?.join()
    }

    @Test
    fun test3() {
        arrayOf(0x802290 to 0x229039).forEach { (c1, c2) ->
            assertEquals(NotificationWindow.combineColors(c1, c2),
                    NotificationWindow.combineColors(c2, c1))
        }
    }

}

fun main(vararg args: String) {
    NotificationTest().test2()
}
